### PR TITLE
Created Ko-Fi presence

### LINKED ISSUE #3811: Ko-Fi | ko-fi.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 

**What's the name of the service?** 

https://ko-fi.com | Ko-Fi

**What should the Presence display?** 

What page a user is on, especially for what Ko-Fi user another is looking at

**If possible, please provide a logo for the service (512x512 minimum)** 

https://uploads-ssl.webflow.com/5c14e387dab576fe667689cf/5ca5bf1dff3c03fbf7cc9b3c_Kofi_logo_RGB_rounded.png

### PR DESCRIPTION
**Pull request by dzlandis**
Ko-Fi Presence, resolves #3811 

**Name of the service:**
Ko-Fi

**Link to the website:**
https://ko-fi.com/

**Screenshots:**
Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img Img

### CODE REVIEW

**Comment by dzlandis** [comment]
Ready to be reviewed 😉

File: websites/K/Ko-Fi/presence.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+          )
+          .textContent.split("Jump to")[0]
+          .trim(),
+        shopOwner = document
+          .querySelector(
+            "#shop-item-detail > div > div.kfds-lyt-between-algn-top-row-to-col.kfds-c-sticky > div.sidebar.kfds-c-sticky-wrapper.kfds-c-order-2.kfds-c-shop-detail-wrapper > div.kfds-lyt-width-100.kfds-c-lyt-pdg-16-24.kfds-c-shop-detail-column-control > div > a > div > span:nth-child(1)"
+          )
+          .textContent.split("Jump to")[0]
+          .trim(),
+        URLSplit = document.location.pathname.split("/"),
```
Comment (Bas950) [inline websites/K/Ko-Fi/presence.ts:81]: use array deconstructing

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/K/Ko-Fi/presence.ts (hunk 1 above)
Comment (dzlandis) [reply websites/K/Ko-Fi/presence.ts:81]: I'm not exactly sure what you mean in this context. Could you please give me an example 🥺

Comment (SlowLife1661) [reply websites/K/Ko-Fi/presence.ts:81]: > use array deconstructing

I think it's fine

Comment (dzlandis) [reply websites/K/Ko-Fi/presence.ts:81]: > I think it's fine

Let me know if I can close this conversation.

Comment (Bas950) [reply websites/K/Ko-Fi/presence.ts:81]: I mean with URLSplit because you only use [2]

Comment (dzlandis) [reply websites/K/Ko-Fi/presence.ts]: So I tried a bunch of things as you can see and nothing worked so if you could please give me some sort of example of what you mean @Bas950 that would be great.

File: websites/K/Ko-Fi/presence.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+          .textContent
+          .trim(),
+        shopOwner = document
+          .querySelector(
+            "#shop-item-detail > div > div.kfds-lyt-between-algn-top-row-to-col.kfds-c-sticky > div.sidebar.kfds-c-sticky-wrapper.kfds-c-order-2.kfds-c-shop-detail-wrapper > div.kfds-lyt-width-100.kfds-c-lyt-pdg-16-24.kfds-c-shop-detail-column-control > div > a > div > span:nth-child(1)"
+          )
+          .textContent
+          .trim(),
+        [ URLSplit ] = document.location.pathname.split("/"),
+        URL = `https://ko-fi.com/s/${URLSplit[2]}?ref=premid_discord_presence`;
```
Comment (Bas950) [inline websites/K/Ko-Fi/presence.ts:82]:
```suggestion
        [, URLSplit ] = document.location.pathname.split("/"),
        URL = `https://ko-fi.com/s/${URLSplit}?ref=premid_discord_presence`;
```

**Review by Bas950 — COMMENTED** [review]
that should work

**Comment by dzlandis** [comment]
I also tested these changes and everything is still working just fine. 👍

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**